File: Java/34. Find First and Last Position of Element in Sorted Array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        if(nums.length == 0){
            return new int[] {-1, -1};
        }
        
        int start = searchStart(nums, target);
        int end = searchEnd(nums, target);
        
        if(start < nums.length && start <= end){
            return new int[] {start, end};
        }else{
            return new int[] {-1, -1};
        }
    }
    
    private int searchStart(int[] nums, int target){
        int left = 0, right = nums.length-1;
        while(left <= right){
            int mid = (right-left)/2 + left;
            if(nums[mid] >= target){
                right = mid-1;
            }else{
                left = mid+1;
            }
        }
        return left;
    }
    
    private int searchEnd(int[] nums, int target){
        int left = 0, right = nums.length-1;
        while(left <= right){
            int mid = (right-left)/2 + left;
            if(nums[mid] > target){
                right = mid-1;
            }else{
                left = mid+1;
            }
        }
        return right;
    }
}

File: Python3/34. Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if not nums:
            return [-1, -1]
        
        start = self.searchStart(nums, target)
        end = self.searchEnd(nums, target)
        
        if start < len(nums) and start <= end and nums[start] == target:
            return [start, end]
        else:
            return [-1, -1]
        
        
    def searchStart(self, nums, target) -> int:
        left, right = 0, len(nums)-1
        while left <= right:
            mid = (right-left)//2+left
            if nums[mid] >= target:
                right = mid-1
            else:
                left = mid+1
        return left
    
    def searchEnd(self, nums, target) -> int:
        left, right = 0, len(nums)-1
        while left <= right:
            mid = (right-left)//2+left
            if nums[mid] > target:
                right = mid-1
            else:
                left = mid+1
        return right
            
        